functions/database/index.js

```diff
@@ -1,173 +1,438 @@
+function manageDatabase(req, res) {
+  // =========================================================================
+  // Set up our Firebase Service Connection
+  // =========================================================================
+  require("dotenv").config();
+  const admin = require("firebase-admin");
+  const functions = require("firebase-functions");
 
-function manageDatabase (req, res) {
-    // =========================================================================
-    // Set up our Firebase Service Connection
-    // =========================================================================
-    require("dotenv").config();
-    // const firebase = require('firebase/app');
-    // require('firebase/firestore');
+  admin.app().delete();
+  // if (!admin.apps.length) {
+  admin.initializeApp(functions.config().firebase);
+  // }
 
-    // let config = {
-    //     apiKey: process.env.FIREBASE_API_KEY,
-    //     authDomain: process.env.FIREBASE_API_AUTH_DOMAIN,
-    //     projectId: process.env.FIREBASE_API_PROJECT_ID
-    // };
+  let db = admin.firestore();
+  const settings = {
+    timestampsInSnapshots: true
+  };
+  db.settings(settings);
 
-    // if (!firebase.apps.length) {
-    //     firebase.initializeApp(config);
-    // }
+  // =========================================================================
+  // Delete all current collections, then add in all the new data
+  // =========================================================================
+  // Fetch list of collectoins
+  db.getCollections()
```

Comment:
This looks like it works, but it could be done a little bit better. For example, you are nesting promises pretty deeply. Also, you have a bunch of identical `catch` statements. With promises, catches bubble up to the very top promise, so unless you are handling your errors differently for each promise (which you aren't), you only need one catch at the very end.